### PR TITLE
Move static network validation to user edits, allow proxy unset

### PR DESCRIPTION
**Pull request by matthewrsj**
* Move static network validation to user edits

  As the user edits the static IP configuration boxes, their input is
  checked in real-time as they type. Once the field they are editing is
  detected as a valid interface/IP address, the field caption turns
  green, notifying the user of a valid configuration. The button to set
  the static configuration does not activate until all required fields
  are valid.

  Moving this logic to before button activation allows the removal of a
  lot of error handling complexity in the handler() function since the
  user is not allowed to press the button until the input is valid.
  Because of this, this patch also includes considerable simplification
  to the NetworkRequirements class.

* Allow user to unset https_proxy
  
  User can unset their previously entered https_proxy on the network
  requirements screen by (intuitively) deleting their entered proxy
  and pressing the set proxy button again.

### CODE REVIEW

**Comment by matthewrsj** [comment]
### Screenshots

edits1
edits2
edits3
edits4

**Comment by rnesius** [comment]
Very nice!  +1

File: ister_test.py [hunk 1]
```diff
@@ -38,10 +38,19 @@
...
 # pylint: disable=W0702
 # Mock functions almost by definition do not make use of their inputs.
 # Fine if they don't use them
 # pylint: disable=W0613
+# We don't need docstrings for many of these methods, especially when they are
+# just mock methods anyways
+# pylint: disable=missing-docstring
+# Don't worry about protected access warnings since we want our unit tests to
+# test those methods
+# pylint: disable=protected-access
```
Comment (bryteise) [inline ister_test.py]: pylint: disable=protected-access is listed twice.

Comment (matthewrsj) [inline ister_test.py]: oops...

Comment (matthewrsj) [inline ister_test.py]: fixed.

File: ister_gui.py [hunk 2]
```diff
@@ -1049,20 +1045,59 @@ def set_static_edits(self):
...
+
+            if 'DNS' in key.caption and self.static_edits[key]:
+                try:
+                    ipaddress.ip_address(self.static_edits[key])
+                    self.dns_e.set_caption(
+                            ('success', '{0:>20}'.format('DNS (optional): ')))
+                except:
+                    self.dns_e.set_caption(
+                            '{0:>20}'.format('DNS (optional): '))
+                    reqd_found -= 1
```
Comment (bryteise) [inline ister_gui.py]: I am somewhat skeptical that the okay count is easier to understand than 4 different boolean variables but we can leave this as it is for now.

File: ister_gui.py [hunk 3]
```diff
@@ -1086,45 +1121,19 @@ def handler(self, config):
...
-            Alert('Error!', self.error).do_alert()
-            self.error = ''
-            return self._action
+            self.static_button.original_widget.original_widget.set_label(
+                    ('ex', 'Set static IP configuration'))
+
+        # proxy settings may have been added during the self.run_ui() step,
+        # copy our config instance back to the config.
+        config = self.config
+        return self._action
```
Comment (matthewrsj) [inline ister_gui.py]: are you suggesting an error color?

Comment (rnesius) [inline ister_gui.py]: If not empty_string and not_valid, color red?

Comment (matthewrsj) [inline ister_gui.py]: I think the success color coupled with the button becoming activated is enough of an indication.

Another idea is to add an icon, like a green check mark, when the field is validated. This takes a little more work than just updating the caption color though, which is why I went with this option.

Comment (matthewrsj) [inline ister_gui.py]: @rnesius I'm not sure I'm a fan of immediate negative feedback when the user starts typing.

Comment (bryteise) [inline ister_gui.py]: So I'm going to merge this as is. We can go back later and change things if we get feedback.

**Pull request merged**